pandas/core/window/rolling.py

```diff
@@ -454,7 +444,7 @@ def _insert_on_column(self, result: "DataFrame", obj: "DataFrame"):
                 # insert at the end
                 result[name] = extra_col
 
-    def _center_window(self, result, window) -> np.ndarray:
+    def _center_window(self, result: np.ndarray, window) -> np.ndarray:
```

Comment:
ATM _prep_values prevents us from doing anything with EAs, but in principle is that something we'd like to support?

Reply:
In the future yes, but I think a major blocker/consideration is that today rolling agg functions only operate on float64 inputs, so that will need to be addressed in tandem.